### PR TITLE
Feature/books parsing

### PR DESCRIPTION
**Pull request by AlisterIgnatius**

### CODE REVIEW

File: lambdas/books/clientapi_test.go [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+	t.Run("get wikitext", func(t *testing.T) {
+		data, err := client.get(testWiki)
+
+		assert.NoError(err)
+		assert.Equal(handlerTestWikitext, string(data))
+	})
+
+	t.Run("get citation", func(t *testing.T) {
+		data, err := client.get(testCitation)
```
Comment (protsack-stephan) [inline lambdas/books/clientapi_test.go]: You should probably use `GetCitoidBook` here.

File: lambdas/books/clientapi_test.go [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+func TestClient(t *testing.T) {
+	assert := assert.New(t)
+	srv := httptest.NewServer(createTestServer())
+	defer srv.Close()
+
+	client := NewClient(srv.URL)
+
+	t.Run("get wikitext", func(t *testing.T) {
+		data, err := client.get(testWiki)
```
Comment (protsack-stephan) [inline lambdas/books/clientapi_test.go]: You should probably use `GetWikitext` here.

File: lambdas/books/clientapi_test.go [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+	t.Run("get citation", func(t *testing.T) {
+		data, err := client.get(testCitation)
+
+		assert.NoError(err)
+		assert.Equal(handlerTestCitation, string(data))
+	})
+
+	t.Run("get citation fail", func(t *testing.T) {
+		_, err := client.get(testBadReq)
```
Comment (protsack-stephan) [inline lambdas/books/clientapi_test.go]: Again if test case is for citoid you should use `GetCitoidBook` method here.

File: lambdas/books/main.go [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+
+func Handler(title string, cl *Client) {
+	books := make([]Book, 0)
+
+	var cbook *CitoidBook
+	var wikitext string
+	var err error
+
+	if wikitext, err = cl.GetWikitext(title); err != nil {
+		logger.Error("error making HTTP request: %w", err)
```
Comment (protsack-stephan) [inline lambdas/books/main.go:96]: You should probably add return statement after this.

File: lambdas/books/main.go [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+			Isbn:      smatch(line, "isbn"),
+			URL:       smatch(line, "url"),
+			Author:    snmatch(line, "author"),
+			Editor:    fmt.Sprintf("%s %s", smatch(line, "first"), smatch(line, "last")),
+		}
+
+		books = append(books, *book)
+
+		if cbook, err = cl.GetCitoidBook(book.Isbn); err != nil {
+			logger.Error("Error getting citoid books: %w", err)
```
Comment (protsack-stephan) [inline lambdas/books/main.go:120]: You should probably add `return` or `continue` statement after this.

File: lambdas/books/main.go [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+		}
+
+		if cbook != nil {
+			book.Numpages = cbook.Numpages
+		}
+
+		f, err := os.Create(fmt.Sprintf("%s.json", fmt.Sprintf("%s_%s", title, book.Title)))
+		if err != nil {
+			logger.Error("Error creating JSON file: %w", err)
```
Comment (protsack-stephan) [inline lambdas/books/main.go:129]: You should probably add `return` or `continue` statement after this.

File: lambdas/books/main.go (hunk 5 above)
Comment (AlisterIgnatius) [reply lambdas/books/main.go:120]: But if we return/continue here we will terminate json creation but we need to create it even if we had 404 from citoid api.

Comment (protsack-stephan) [reply lambdas/books/main.go:120]: Ok, yeah you're totally right my bad.

**Pull request merged**